electrum/wizard.py

```
@@ -0,0 +1,319 @@
+            return l
+        else:
+            raise Exception('last handler for view {view} is not callable nor a bool literal')
+
+    def finished(self, wizard_data):
+        self._logger.debug('finished.')
+
+    def reset(self):
+        self.stack = []
+        self._current = WizardViewState(None, {}, {})
+
+class NewWalletWizard(AbstractWizard):
+
+    _logger = get_logger(__name__)
+
+    def __init__(self, daemon):
+        self.navmap = {
+            'wallet_name': {
+                'next': 'wallet_type'
+            },
+            'wallet_type': {
+                'next': self.on_wallet_type
+            },
+            'keystore_type': {
+                'next': self.on_keystore_type
+            },
+            'create_seed': {
+                'next': 'confirm_seed'
+            },
+            'confirm_seed': {
+                'next': 'wallet_password',
+                'last': self.last_if_single_password
+            },
+            'have_seed': {
+                'next': self.on_have_seed,
+                'last': self.last_if_single_password_and_not_bip39
+            },
+            'bip39_refine': {
+                'next': 'wallet_password',
+                'last': self.last_if_single_password
+            },
+            'have_master_key': {
+                'next': 'wallet_password',
+                'last': self.last_if_single_password
+            },
+            'imported': {
+                'next': 'wallet_password',
+                'last': self.last_if_single_password
+            },
+            'wallet_password': {
+                'last': True
+            }
+        }
+        self._daemon = daemon
+
+    def start(self, initial_data = {}):
+        self.reset()
+        self._current = WizardViewState('wallet_name', initial_data, {})
+        return self._current
+
+    def last_if_single_password(self, view, wizard_data):
+        return False # TODO: self._daemon.config.get('single_password')
```

Please raise a `NotImplementedError` exception here, instead of returning `False`.
it took me a while to understand that this method must be overloaded.

---

The idea was to provide a common implementation that optionally can be overridden, but as we need to combine the password check, config setting and check whether the other wallets use the same password, I guess this will always need a gui specific implementation.. Will modify to `NotimplementedError`.